### PR TITLE
Chore: enable a modified version of multiline-comment-style rule on codebase

### PR DESCRIPTION
**Pull request by not-an-aardvark**
**What is the purpose of this pull request? (put an "X" next to item)**

[x] Other, please explain:

**What changes did you make? (Give an overview)**

This makes the following changes:

* Creates a project-specific `internal-multiline-comment-style` rule, which is the same as the built-in `multiline-comment-style` rule except that it ignores the banner comments that we use at the top of files. This uses [eslint-rule-composer](https://github.com/not-an-aardvark/eslint-rule-composer) to build off of the existing rule, rather than copying it and making changes.
* Enables the `internal-multiline-comment-style` rule on the ESLint codebase
* Updates a few tests (namely: `tests/lib/cli-engine.js` and `tests/bin/eslint.js`) that rely on linting the CWD, to account for the fact that the config in the CWD now has a project-specific rule.

**Is there anything you'd like reviewers to focus on?**

Nothing in particular

### CODE REVIEW

File: tests/bin/eslint.js [hunk 1]
```diff
@@ -141,10 +141,14 @@ describe("bin/eslint.js", () => {
...
-        it("has exit code 1 if a linting error is reported", () => assertExitCode(runESLint(["bin/eslint.js", "--env", "es6", "--no-eslintrc", "--rule", "semi: [2, never]"]), 1));
-        it("has exit code 1 if a syntax error is thrown", () => assertExitCode(runESLint(["README.md"]), 1));
+        it("has exit code 0 if no linting errors occur", () =>
+            assertExitCode(runESLint(["bin/eslint.js", "--rulesdir=tools/internal-rules"]), 0));
+        it("has exit code 0 if a linting warning is reported", () =>
+            assertExitCode(runESLint(["bin/eslint.js", "--env", "es6", "--no-eslintrc", "--rule", "semi: [1, never]"]), 0));
+        it("has exit code 1 if a linting error is reported", () =>
+            assertExitCode(runESLint(["bin/eslint.js", "--env", "es6", "--no-eslintrc", "--rule", "semi: [2, never]"]), 1));
+        it("has exit code 1 if a syntax error is thrown", () =>
+            assertExitCode(runESLint(["README.md", "--rulesdir=tools/internal-rules"]), 1));
```
Comment (platinumazure) [inline tests/bin/eslint.js]: Why did you change these lines?

Comment (not-an-aardvark) [reply tests/bin/eslint.js]: These test cases rely on ESLint linting itself. I added `--rulesdir=tools/internal-rules` to the second test case to ensure that the config file is valid for that run. The goal is to ensure that the exit code is 1 as a result of the syntax error, not as a result of the missing rule.

**Comment by Aladdin-ADD** [comment]
it has a conflict. @not-an-aardvark

**Comment by not-an-aardvark** [comment]
Fixed the conflict.

File: tools/internal-rules/multiline-comment-style.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+const ruleComposer = require("eslint-rule-composer");
+const multilineCommentStyle = require("../../lib/rules/multiline-comment-style");
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+// The `no-invalid-meta` internal rule has a false positive here.
+// eslint-disable-next-line rulesdir/no-invalid-meta
```
Comment (Aladdin-ADD) [inline tools/internal-rules/multiline-comment-style.js]: unrelated: no-invalid-meta seems ideal to be a rule in `eslint-plugin-eslint-plugin`, so that other plugin authors can use easily.

File: lib/token-store/backward-token-cursor.js [hunk 3]
```diff
@@ -45,9 +45,11 @@ module.exports = class BackwardTokenCursor extends Cursor {
...
     }
 
-    //
-    // Shorthand for performance.
-    //
+    /*
+     *
+     * Shorthand for performance.
+     *
+     */
```
[the remark this answers is not in the document]
Comment (not-an-aardvark) [reply lib/token-store/backward-token-cursor.js]: It was created by the autofixer. The [original comment](https://github.com/eslint/eslint/blob/cb60285b57066010eb439f4b70bc792a4f24e506/lib/token-store/backward-token-cursor.js#L48-L50) also contained blank lines, so I think it's fine that the updated comment has them as well.
